ChemistryLib/CreateChemicalSolverInterface.cpp

```diff
@@ -0,0 +1,137 @@
+        config.getConfigSubtree("solution"),
+        process_id_to_component_name_map);
+
+    // kinetic reactants
+    auto kinetic_reactants = PhreeqcIOData::createKineticReactants(
+        //! \ogs_file_param{prj__chemical_system__kinetic_reactants}
+        config.getConfigSubtreeOptional("kinetic_reactants"), mesh);
+
+    // rates
+    auto reaction_rates = createReactionRates<PhreeqcIOData::ReactionRate>(
+        //! \ogs_file_param{prj__chemical_system__rates}
+        config.getConfigSubtreeOptional("rates"));
+
+    auto const num_chemical_systems = mesh.getNumberOfBaseNodes();
+    std::vector<PhreeqcIOData::AqueousSolution> aqueous_solutions(
+        num_chemical_systems, aqueous_solution);
+
+    // equilibrium phases
+    auto equilibrium_phases = PhreeqcIOData::createEquilibriumPhases(
+        //! \ogs_file_param{prj__chemical_system__equilibrium_phases}
+        config.getConfigSubtreeOptional("equilibrium_phases"), mesh);
+
+    // output
+    auto const& components = aqueous_solution.components;
+    auto const project_file_name = BaseLib::joinPaths(
+        output_directory,
+        BaseLib::extractBaseNameWithoutExtension(config.getProjectFileName()));
+    auto output = PhreeqcIOData::createOutput(
+        components, equilibrium_phases, kinetic_reactants, project_file_name);
+
+    return std::make_unique<PhreeqcIOData::PhreeqcIO>(
+        project_file_name, std::move(path_to_database),
+        std::move(aqueous_solutions), std::move(equilibrium_phases),
+        std::move(kinetic_reactants), std::move(reaction_rates),
+        std::move(output), process_id_to_component_name_map);
+}
+
+template <>
+std::unique_ptr<ChemicalSolverInterface>
+createChemicalSolverInterface<ChemicalSolver::PhreeqcKernel>(
+    MeshLib::Mesh const& mesh,
+    std::vector<std::pair<int, std::string>> const&
+        process_id_to_component_name_map,
+    BaseLib::ConfigTree const& config, std::string const& /*output_directory*/)
+{
+    auto path_to_database = parseDatabasePath(config);
+
+    // solution
+    auto aqueous_solution = PhreeqcKernelData::createAqueousSolution(
+        //! \ogs_file_param{prj__chemical_system__solution}
+        config.getConfigSubtree("solution"),
+        process_id_to_component_name_map);
+
+    // kinetic reactants
+    auto kinetic_reactants = PhreeqcKernelData::createKineticReactants(
+        //! \ogs_file_param{prj__chemical_system__kinetic_reactants}
+        config.getConfigSubtreeOptional("kinetic_reactants"), mesh);
+
+    // rates
+    auto reaction_rates = createReactionRates<PhreeqcKernelData::ReactionRate>(
+        //! \ogs_file_param{prj__chemical_system__rates}
+        config.getConfigSubtreeOptional("rates"));
+
+    return std::make_unique<PhreeqcKernelData::PhreeqcKernel>(
+        mesh.getNumberOfBaseNodes(), process_id_to_component_name_map,
+        path_to_database, aqueous_solution, std::move(kinetic_reactants),
+        std::move(reaction_rates));
```

Comment:
Is it worth to `std::move` the `path_to_database` and `aqueous_solution`? Or is there any reason not to do so?

Reply:
Enclosed in `std::move`.